source/extensions/common/wasm/wasm.cc

```diff
@@ -806,12 +806,10 @@ bool Context::isSsl() { return decoder_callbacks_->connection()->ssl() != nullpt
 // Calls into the WASM code.
 //
 void Context::onStart() {
+  wasm_->wasmVm()->start(this);
```

Comment:
Any test to cover the behavior?

Reply:
`ensuring that globals are initialized`. I get it. Thanks!

Reply:
This adds -O3 to a WASM file for a test that used to cause a crash with -O3, so that is the test.